crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -188,29 +208,53 @@ impl ContainerBuilder {
         self
     }
 
+    /// This function only adds the [`VolumeMount`] in case there is no volumeMount with the same mountPath already.
+    /// In case there already was a volumeMount with the same path already, an [`tracing::error`] is raised in case the
+    /// contents of the volumeMounts differ.
+    ///
+    /// Historically this function unconditionally added volumeMounts, which resulted in invalid
+    /// [`k8s_openapi::api::core::v1::PodSpec`]s, as volumeMounts where added multiple times - think of Trino using the same [`crate::commons::s3::S3Connection`]
+    /// two times, resulting in e.g. the s3 credentials being mounted twice as the same volumeMount.
+    pub fn add_volume_mount_struct(&mut self, volume_mount: VolumeMount) -> Result<&mut Self> {
+        if let Some(existing_volume_mount) = self.volume_mounts.get(&volume_mount.mount_path) {
+            ensure!(
+                existing_volume_mount.eq(&volume_mount),
```

Comment:
question: Any particular reason why we don't use `existing_volume_mount == volume_mount` here?

Reply:
Nope, my fault ;) 1a3f95f